src/Market.ts

```diff
@@ -33,14 +33,20 @@ export class Market implements MarketProperties {
     return cost;
   }
 
-  getResourceCostForOpenSlots(openSlots: number): number {
-    if (openSlots > this.maxOpenSlots) {
-      openSlots = this.maxOpenSlots;
+  getResourceBenefit(count: number): number {
+    let benefit = 0;
```

Comment:
Declare variables as close to use as possible, so that it's easier to understand how they're used. Move this just above the for loop